monitoring/src/main/java/io/silverware/microservices/monitoring/MetricsManager.java

```diff
@@ -0,0 +1,211 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------/
+ */
+package io.silverware.microservices.monitoring;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.LongAdder;
+
+/**
+ * Class responsible for managing metrics. Class works with {@link DefaultMetric} and stores recorded values for one minute. Percentiles are updated every second in separate thread.
+ */
+public class MetricsManager {
+
+   private final Metrics metrics = new DefaultMetric();
+
+   private final LoadingCache<Long, BigDecimal> mapValues = CacheBuilder.newBuilder().expireAfterWrite(1, TimeUnit.MINUTES).build(new CacheLoader<Long, BigDecimal>() {
+      @Override
+      public BigDecimal load(Long key) throws Exception {
+         return mapValues.get(key);
+      }
+   });
+
+   private LongAdder longAdder = new LongAdder();
+
+   private final Thread updateThread = new Thread(this::updatePercentiles);
+
+   /**
+    * Default constructor, sets true for daemon attribute of percentile updating thread and starts it.
+    */
+   public MetricsManager() {
+      updateThread.setDaemon(true);
+      updateThread.start();
+   }
+
+   /**
+    * Method responsible for adding new time to values collection and also updating min, max, avg and count metrics.
+    *
+    * @param elapsedTime runtime of microservice method.
+    */
+   public void addTime(BigDecimal elapsedTime) {
+
+      if (elapsedTime.compareTo(BigDecimal.ZERO) <= 0) {
+         throw new IllegalArgumentException("Elapsed time is negative or zero");
+      }
+
+      mapValues.put(longAdder.longValue(), elapsedTime);
+
+      longAdder.increment();
+
+      final BigDecimal count = new BigDecimal(metrics.getCount());
+      final BigDecimal averageTime = metrics.getAverageTime();
+      final BigDecimal minTime = metrics.getMinTime();
+      final BigDecimal maxTime = metrics.getMaxTime();
+
+      metrics.incrementCount();
+
+      metrics.setAverageTime((averageTime.multiply(count).add(elapsedTime)).divide(count.add(BigDecimal.ONE), BigDecimal.ROUND_HALF_UP));
+
+      if (elapsedTime.compareTo(maxTime) == 1) {
+         metrics.setMaxTime(elapsedTime);
+      }
+
+      if (elapsedTime.compareTo(minTime) == -1 || metrics.getMinTime().equals(BigDecimal.ZERO)) {
```

Comment:
This condition is not necessary. There are no more results of `BigDecimal.compareTo` method other than `-1`, `0` or `1`. If we want to be 100% sure, the previous condition should be `> 1`.